sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_response_handlers.py

```diff
@@ -193,36 +192,54 @@ def healthcare_result(health_result, results, *args, **kwargs): # pylint: disabl
     return AnalyzeHealthcareResultItem._from_generated(health_result) # pylint: disable=protected-access
 
 
-def analyze_result(doc_id_order, obj, response_headers, tasks, **kwargs): # pylint: disable=unused-argument
-    return TextAnalysisResult(
-        entities_recognition_results=[
-            EntitiesRecognitionTaskResult(
-                name=t.name,
-                results=entities_result(doc_id_order, t.results, response_headers, lro=True)
-            ) for t in tasks.entity_recognition_tasks
-        ] if tasks.entity_recognition_tasks else [],
-        pii_entities_recognition_results=[
-            PiiEntitiesRecognitionTaskResult(
-                name=t.name,
-                results=pii_entities_result(doc_id_order, t.results, response_headers, lro=True)
-            ) for t in tasks.entity_recognition_pii_tasks
-        ] if tasks.entity_recognition_pii_tasks else [],
-        key_phrase_extraction_results=[
-            KeyPhraseExtractionTaskResult(
-                name=t.name,
-                results=key_phrases_result(doc_id_order, t.results, response_headers, lro=True)
-            ) for t in tasks.key_phrase_extraction_tasks
-        ] if tasks.key_phrase_extraction_tasks else []
-    )
-
-
 def healthcare_extract_page_data(doc_id_order, obj, response_headers, health_job_state): # pylint: disable=unused-argument
     return (health_job_state.next_link,
         healthcare_result(doc_id_order, health_job_state.results, response_headers, lro=True))
 
+def _get_deserialization_callback_from_task_type(task_type):
+    if task_type == AnalyzeBatchActionsType.RECOGNIZE_ENTITIES:
+        return entities_result
+    if task_type == AnalyzeBatchActionsType.RECOGNIZE_PII_ENTITIES:
+        return pii_entities_result
+    return key_phrases_result
+
+def _get_property_name_from_task_type(task_type):
+    if task_type == AnalyzeBatchActionsType.RECOGNIZE_ENTITIES:
+        return "entity_recognition_tasks"
+    if task_type == AnalyzeBatchActionsType.RECOGNIZE_PII_ENTITIES:
+        return "entity_recognition_pii_tasks"
+    return "key_phrase_extraction_tasks"
+
+def _num_tasks_in_current_page(returned_tasks_object):
+    return (
+        len(returned_tasks_object.entity_recognition_tasks or []) +
+        len(returned_tasks_object.entity_recognition_pii_tasks or []) +
+        len(returned_tasks_object.key_phrase_extraction_tasks or [])
+    )
 
-def analyze_extract_page_data(doc_id_order, obj, response_headers, analyze_job_state):
-    return analyze_job_state.next_link, [analyze_result(doc_id_order, obj, response_headers, analyze_job_state.tasks)]
+def get_iter_items(doc_id_order, task_order, response_headers, analyze_job_state):
```

Comment:
basically, I pass in the order for the different action results users inputted. Then, use that to ensure the output order of task results is in the same result as input

Reply:
will this work if the user sent more of the same action? i.e. 2 piientities actions with different arguments?